js/main.js

```diff
@@ -1,2 +1,34 @@
 /* global data */
 /* exported data */
+
+var photoURL = document.querySelector('.photourl');
+var photo = document.querySelector('img');
+var form = document.querySelector('.form');
+var title = document.querySelector('.title');
+var notes = document.querySelector('.notes');
+
+photoURL.addEventListener('input', updateImage);
+form.addEventListener('submit', submit);
+
+function updateImage(event) {
+  photo.setAttribute('src', photoURL.value);
+}
+
+function submit(event) {
+  event.preventDefault();
+  var object = { Title: title.value, URL: photoURL.value, Notes: notes.value };
+  object.nextEntryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.unshift(object);
+  photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  form.reset();
+  var dataEntriesJSON = JSON.stringify(data.entries);
+  localStorage.setItem('data-entry', dataEntriesJSON);
+}
+
+var previousEntriesJSON = localStorage.getItem('data-entry');
+if (previousEntriesJSON !== null) {
+  data.entries.push(JSON.parse(previousEntriesJSON));
+  data.entries.flat();
+  data.nextEntryId = data.entries.length + 1;
+}
```

Comment:
This should be being done in `data.js`, not in `main.js`. Also, you should save the entire `data` object in local storage, not just the `entries` array